connection.go

```diff
@@ -29,30 +29,40 @@ type StreamHandler func(stream *Stream)
 type AuthHandler func(header http.Header, slot uint8, parent uint32) bool
 
 type idleAwareFramer struct {
-	f       *spdy.Framer
-	conn    *Connection
-	expired *time.Timer
-	startCh chan struct{}
+	f              *spdy.Framer
+	conn           *Connection
+	expired        *time.Timer
```

Comment:
Also, you get a nil pointer dereference error if 1 of the case statements tries to read from a nil channel